Issue While Containerizing.

I am creating a server-side application and using a truffle setup to communicate to the blockchain. Locally the project works fine but if I try to containerize the project truffle build fails.

## Expected Behavior

The installation should complete easily and I should have a running container.

## Actual Results

```
`npm WARN deprecated mkdirp@0.5.1: Legacy versions of mkdirp are no longer supported. Please update to mkdirp 1.x. (Note that the API surface has changed to use Promises in 1.x.)
npm WARN tar ENOENT: no such file or directory, open '/build/node_modules/.staging/xhr2-cookies-2a921703/README.md'
npm WARN tar ENOENT: no such file or directory, lstat '/build/node_modules/.staging/prettier-plugin-solidity-7f4b2c31/tests/InheritanceSpecifier/__snapshots__'
npm WARN tar ENOENT: no such file or directory, open '/build/node_modules/.staging/eth-tx-summary-f333bb5e/example.js'
npm WARN tar ENOENT: no such file or directory, open '/build/node_modules/.staging/eth-tx-summary-f333bb5e/expected-trace.json'
npm WARN tar ENOENT: no such file or directory, open '/build/node_modules/.staging/@solidity-parser/parser-11454a72/dist/antlr4/atn/PredictionMode.js'
npm WARN tar ENOENT: no such file or directory, lstat '/build/node_modules/.staging/prettier-plugin-solidity-7f4b2c31/tests/MemberAccess/__snapshots__'
npm WARN tar ENOENT: no such file or directory, lstat '/build/node_modules/.staging/prettier-plugin-solidity-7f4b2c31/tests/ModifierDefinitions/__snapshots__'
npm WARN tar ENOENT: no such file or directory, open '/build/node_modules/.staging/@openzeppelin/contracts-d24218d3/build/contracts/ERC20Burnable.json'
npm WARN tar ENOENT: no such file or directory, lstat '/build/node_modules/.staging/prettier-plugin-solidity-7f4b2c31/tests/NameValueExpression/__snapshots__'
npm WARN tar ENOENT: no such file or directory, lstat '/build/node_modules/.staging/prettier-plugin-solidity-7f4b2c31/tests/Ownable/__snapshots__'
npm WARN tar ENOENT: no such file or directory, open '/build/node_modules/.staging/mocha-65c59efb/README.md'
npm WARN tar ENOENT: no such file or directory, lstat '/build/node_modules/.staging/prettier-plugin-solidity-7f4b2c31/tests/Pragma/__snapshots__'
npm WARN tar ENOENT: no such file or directory, lstat '/build/node_modules/.staging/prettier-plugin-solidity-7f4b2c31/tests/PrettierIgnore/__snapshots__'
npm WARN tar ENOENT: no such file or directory, lstat '/build/node_modules/.staging/prettier-plugin-solidity-7f4b2c31/tests/Proxy/__snapshots__'
npm WARN tar ENOENT: no such file or directory, open '/build/node_modules/.staging/ethereumjs-block-eb5aba83/tests/testdata-from-rpc.json'
npm WARN tar ENOENT: no such file or directory, open '/build/node_modules/.staging/ethereumjs-block-eb5aba83/tests/testdata.json'
npm WARN tar ENOENT: no such file or directory, open '/build/node_modules/.staging/ethereumjs-block-eb5aba83/tests/testdata2.json'
npm WARN tar ENOENT: no such file or directory, lstat '/build/node_modules/.staging/prettier-plugin-solidity-7f4b2c31/tests/quotes/__snapshots__'
npm WARN tar ENOENT: no such file or directory, lstat '/build/node_modules/.staging/prettier-plugin-solidity-7f4b2c31/tests'
npm WARN tar ENOENT: no such file or directory, open '/build/node_modules/.staging/ethers-83107a5a/dist/ethers.min.js'
npm WARN tar ENOENT: no such file or directory, open '/build/node_modules/.staging/typescript-bf155e1e/lib/tsserverlibrary.d.ts'
npm WARN tar ENOENT: no such file or directory, open '/build/node_modules/.staging/truffle-6bd01753/test/scenarios/memorylogger.js'
npm WARN tar ENOENT: no such file or directory, open '/build/node_modules/.staging/truffle-6bd01753/test/sources/external_compile/test/metacoin.js'
npm WARN tar ENOENT: no such file or directory, open '/build/node_modules/.staging/truffle-6bd01753/test/scenarios/commands/migrate.js'
npm WARN tar ENOENT: no such file or directory, open '/build/node_modules/.staging/truffle-6bd01753/test/scenarios/migrations/migrate.js'
npm WARN tar ENOENT: no such file or directory, open '/build/node_modules/.staging/truffle-6bd01753/build/templates/migration.js'
npm WARN tar ENOENT: no such file or directory, open '/build/node_modules/.staging/truffle-6bd01753/test/scenarios/commands/networks.js'
npm ERR! code ENOENT
npm ERR! syscall spawn git
npm ERR! path git
npm ERR! errno ENOENT
npm ERR! enoent Error while executing:
npm ERR! enoent undefined ls-remote -h -t ssh://git@github.com/web3-js/WebSocket-Node.git
npm ERR! enoent 
npm ERR! enoent 
npm ERR! enoent spawn git ENOENT
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent 
npm WARN tar ENOENT: no such file or directory, lstat '/build/node_modules/.staging/truffle-6bd01753/test/scenarios'
npm WARN tar ENOENT: no such file or directory, lstat '/build/node_modules/.staging/truffle-6bd01753/test/scenarios'
```

## Environment

Operating System: Ubunto
Truffle v5.1.29 (core: 5.1.29)
Solidity - 0.6.2 (solc-js)
Node v10.19.0
Web3.js v1.2.1

@rachit2501 Have you tried updating `@truffle/hdwallet-provider` and `truffle` to latest?

Am currently troubleshooting and after updating am getting this error:

```
npm ERR! code ENOENT
npm ERR! syscall spawn git
npm ERR! path git
npm ERR! errno ENOENT
npm ERR! enoent Error while executing:
npm ERR! enoent undefined ls-remote -h -t https://github.com/ethereumjs/ethereumjs-abi.git
npm ERR! enoent 
npm ERR! enoent 
npm ERR! enoent spawn git ENOENT
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent 
```

Look like a nested `git` dependency issue coming from `eth-sig-util`: https://github.com/MetaMask/eth-sig-util/blob/1.x/package.json#L24

--- a/packages/hdwallet-provider/package.json
+++ b/packages/hdwallet-provider/package.json
@@ -17,7 +17,7 @@
   },
   "types": "dist/index.d.ts",
   "dependencies": {
-    "@trufflesuite/web3-provider-engine": "15.0.13-0",
+    "@trufflesuite/web3-provider-engine": "15.0.13-1",
     "@types/web3": "^1.0.20",
     "any-promise": "^1.3.0",
     "bindings": "^1.5.0",

